index.js

```diff
@@ -1,21 +1,28 @@
-const GraphQLAdapter = require('./lib')
-const cds = require ('@sap/cds')
-const { decodeURIComponent } = cds.utils
+const cds = require('@sap/cds')
 const LOG = cds.log('graphql')
-const express = require ('express')
+const express = require('express')
+const { createHandler } = require('graphql-http/lib/use/express')
+const { generateSchema4 } = require('./lib/schema')
+const graphiql = require('./app/graphiql')
+const { decodeURIComponent } = cds.utils
 
-function CDSGraphQLAdapter (options) {
-  const {services} = options
+function GraphQLAdapter(options) {
+  const router = express.Router()
+  const { services } = options
   const defaults = { graphiql: true }
+  const schema = generateSchema4(services)
   options = { ...defaults, ...options }
 
-  return express.Router()
-  .use (express.json()) //> required by logger below
-  .use ((req,_,next)=>{
-    LOG.info (req.method, req.body?.query || decodeURIComponent(req.query.query))
-    next()
-  })
-  .use (new GraphQLAdapter (services, options))
+  router
+    .use(express.json()) //> required by logger below
+    .use((req, _, next) => {
+      LOG.info(req.method, req.body?.query || decodeURIComponent(req.query.query))
+      next()
+    })
+
+  if (options.graphiql) router.get('/', graphiql)
+  router.use((req, res) => createHandler({ schema, context: { req, res }, ...options })(req, res))
+  return router
 }
 
-module.exports = CDSGraphQLAdapter
+module.exports = GraphQLAdapter
```

Comment:
No need to not merge this, but maybe we should clarify if @danjoa prefers `GraphQLAdapter` or `CDSGraphQLAdapter` here.